src/GovUk.Education.ExploreEducationStatistics.Admin/Security/AuthorizationHandlers/AssignPrereleaseContactsToSpecificReleaseAuthorizationHandlers.cs

```diff
@@ -37,19 +33,14 @@ public class AssignPrereleaseContactsToSpecificReleaseAuthorizationHandler
                 context.Succeed(requirement);
                 return;
             }
-
-            var publicationRoles =
-                await _userPublicationRoleRepository.GetAllRolesByUserAndPublication(context.User.GetUserId(), release.PublicationId);
-
-            if (ContainPublicationOwnerRole(publicationRoles))
-            {
-                context.Succeed(requirement);
-                return;
-            }
             
-            var releaseRoles = await _userReleaseRoleRepository.GetDistinctRolesByUserAndRelease(context.User.GetUserId(), release.Id);
-
-            if (ContainsEditorOrApproverRole(releaseRoles))
+            if (await _authorizationHandlerResourceRoleService
+                    .HasRolesOnPublicationOrRelease(
+                        context.User.GetUserId(),
+                        release.PublicationId,
+                        release.Id,
+                        ListOf(PublicationRole.Owner),
```

Comment:
Should PublicationRole.ReleaseApprover be included here, since ReleaseRole.Approver is included?